### PR TITLE
Sort sitemaps alphabetically

### LINKED ISSUE #270: Order sitemaps alphabetically
Especially in the app, sitemaps suggest themselves as the primary unit of order. Unfortunately, the order of sitemaps in the sidebar seems to be absolutely random with no possibility to influence it. Would it be possible to order the sitemaps alphabetically, at least?

### PR DESCRIPTION
**Pull request by mueller-ma**
but let the default sitemap always be the first one.

Fixes #270

### CODE REVIEW

**Comment by mueller-ma** [comment]
@maniac103 May I ask for a review?

File: mobile/src/main/java/org/openhab/habdroid/ui/OpenHABMainActivity.java [hunk 1]
```diff
@@ -784,10 +784,13 @@ public void onSuccess(Call call, int statusCode, Headers headers, byte[] respons
...
 
+                String defaultSitemapName =
+                        mSettings.getString(Constants.PREFERENCE_SITEMAP_NAME, "");
+
                 // OH1 returns XML, later versions return JSON
                 List<OpenHABSitemap> result = mOpenHABVersion == 1
-                        ? loadSitemapsFromXml(responseBody)
-                        : loadSitemapsFromJson(responseBody);
+                        ? loadSitemapsFromXml(responseBody, defaultSitemapName)
+                        : loadSitemapsFromJson(responseBody, defaultSitemapName);
```
Comment (maniac103) [inline mobile/src/main/java/org/openhab/habdroid/ui/OpenHABMainActivity.java]: I don't like default sitemap name being passed to the load methods, as it's not required for loading. Suggestion: leave sorting out of the load methods completely and instead ...

File: mobile/src/main/java/org/openhab/habdroid/ui/OpenHABMainActivity.java [hunk 2]
```diff
@@ -784,10 +784,13 @@ public void onSuccess(Call call, int statusCode, Headers headers, byte[] respons
...
                 Log.d(TAG, "Server returned sitemaps: " + result);
                 mSitemapList.clear();
                 if (result != null) {
```
Comment (maniac103) [inline mobile/src/main/java/org/openhab/habdroid/ui/OpenHABMainActivity.java]: ... sort `result` here (or `mSitemapList` below).

Separating those also makes for better unit testing.

**Comment by mueller-ma** [comment]
> I'm not sure whether we should reorder the currently selected sitemap to the top at all

It's the default sitemap, not the currently open one.

**Comment by maniac103** [comment]
> It's the default sitemap, not the currently open one. I think it's a good idea, since it is for the primary sitemap.

Fair enough. Fine with it then, I'd just (strongly) prefer splitting out loading and sorting of the sitemaps.

**Comment by mueller-ma** [comment]
> Separating those also makes for better unit testing.

Indeed, much better for testing.

**Pull request merged**